### PR TITLE
Generalise HA Specific config.

### PR DESCRIPTION
**Pull request by apdibbo**
Add functions to create a uri/url from a dictionary of hosts and to get the controller host

Move to using dictionaries for defining all hosts so that there isnt a change of variable if you move to High Availability
Tweaks to generalise ha specific config

### CODE REVIEW

File: defaults/openstack/config.pan [hunk 1]
```diff
@@ -974,107 +1006,54 @@ final variable OPENSTACK_CEPH_LIBVIRT_SECRET ?= {
...
-  default = OPENSTACK_MEMCACHE_HOST
-  note = This is a list of hosts to be used for Memcached
-}
-final variable OPENSTACK_MEMCACHE_HOSTS ?= dict(OPENSTACK_MEMCACHE_HOST,OPENSTACK_MEMCACHE_PORT);
-@use{
   type = ipaddress
   note = The floating ip address to use for keepalived
 }
-final variable OPENSTACK_FLOATING_IP ?= if (OPENSTACK_HA) {error('OPENSTACK_FLOATING_IP must be set for high availability');} else {null;};
+variable OPENSTACK_FLOATING_IP ?= if (OPENSTACK_HA) {error('OPENSTACK_FLOATING_IP must be set for high availability');} else {null;};
```
Comment (guillaume-philippon) [inline defaults/openstack/config.pan]: FLOATING_IP can mislead as it use on OPENSTACK too for a functionnality OPENSTACK_VIRTUAL_IP or OPENSTACK_KEEPALIVE_IP could be better.

**Review by guillaume-philippon — COMMENTED** [review]
Seems great for me except for keeplive IP variable name.

If it s deploy @RAL we could merge it for me.

**Comment by apdibbo** [comment]
This is deployed at RAL and I have updated the variable name as suggested

**Comment by jrha** [comment]
Can you squash the typos fixes into the commits that added the typos please.

**Comment by apdibbo** [comment]
Squashed

**Pull request merged**